cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Broker.java

```diff
@@ -682,7 +688,7 @@ public void writeTo(OutputStream out) throws IOException {
     }
     // If information of replica placement over disk is not populated, write replica information to output stream;
     // otherwise disk will write replica information to output stream.
-    if (_diskByLogdir.values().stream().mapToInt(disk -> disk.replicas().size()).sum() == 0) {
+    if (!_replicaPlacementInfoPopulated) {
```

Comment:
Isn't it enough to check `if (_diskByLogdir.isEmpty()) {`? If so, we can remove `_replicaPlacementInfoPopulated` completely.

Reply:
Somehow I am unable to reply your other comment about `_replicaPlacementInfoPopulated`, I will reply all here.
The `_replicaPlacementInfoPopulated` here is mainly to distinguish following scenarios(1) and (4).

(1) [`BrokerCapacityConfigResolver`](https://github.com/linkedin/cruise-control/blob/8ba6410e24d465b873e2c5f999bb220426654076/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java#L512) reports disk capacity(i.e `_diskByLogdir` is not empty), but [`populateReplicaPlacementInfo` ](https://github.com/linkedin/cruise-control/blob/8ba6410e24d465b873e2c5f999bb220426654076/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java#L521)is false 
-> 
[`diskUtils()`](https://github.com/linkedin/cruise-control/blob/8ba6410e24d465b873e2c5f999bb220426654076/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Broker.java#L649) return empty map and [`diskCapacities()`](https://github.com/linkedin/cruise-control/blob/8ba6410e24d465b873e2c5f999bb220426654076/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Broker.java#L668) returns capacity map.

(2) `BrokerCapacityConfigResolver` does not report disk capacity(i.e _diskByLogdir is empty), but `populateReplicaPlacementInfo` is false
->
`diskUtils()` return empty map and `diskCapacities()` returns empty map.

(3) `BrokerCapacityConfigResolver` does not report disk capacity(i.e _diskByLogdir is empty), but `populateReplicaPlacementInfo` is true
->
unsupported

(4) `BrokerCapacityConfigResolver` reports disk capacity(i.e _diskByLogdir is not empty), and `populateReplicaPlacementInfo` is true
->
`diskUtils()` return utils map and `diskCapacities()` returns capacity map.